tests/test_frame.py

```diff
@@ -3448,41 +3449,72 @@ def test_groupby_apply() -> None:
     def sum_mean(x: pd.DataFrame) -> float:
         return x.sum().mean()
 
-    with pytest_warns_bounded(
-        DeprecationWarning,
-        "DataFrameGroupBy.apply operated on the grouping columns.",
-        upper="2.99",
+    with (
+        pytest_warns_bounded(
+            DeprecationWarning,
+            "DataFrameGroupBy.apply operated on the grouping columns.",
+            upper="2.2.99",
+        ),
+        pytest_warns_bounded(
+            FutureWarning,
+            "DataFrameGroupBy.apply operated on the grouping columns.",
+            lower="2.2.99",
+            upper="2.3.99",
+        ),
     ):
         check(
             assert_type(df.groupby("col1").apply(sum_mean), pd.Series),
             pd.Series,
         )
 
     lfunc: Callable[[pd.DataFrame], float] = lambda x: x.sum().mean()
-    with pytest_warns_bounded(
-        DeprecationWarning,
-        "DataFrameGroupBy.apply operated on the grouping columns.",
-        upper="2.99",
+    with (
+        pytest_warns_bounded(
+            DeprecationWarning,
+            "DataFrameGroupBy.apply operated on the grouping columns.",
+            upper="2.2.99",
+        ),
+        pytest_warns_bounded(
+            FutureWarning,
+            "DataFrameGroupBy.apply operated on the grouping columns.",
+            lower="2.2.99",
+        ),
     ):
         check(assert_type(df.groupby("col1").apply(lfunc), pd.Series), pd.Series)
 
     def sum_to_list(x: pd.DataFrame) -> list:
         return x.sum().tolist()
 
-    with pytest_warns_bounded(
-        DeprecationWarning,
-        "DataFrameGroupBy.apply operated on the grouping columns.",
-        upper="2.99",
+    with (
+        pytest_warns_bounded(
+            DeprecationWarning,
+            "DataFrameGroupBy.apply operated on the grouping columns.",
+            upper="2.2.99",
+        ),
+        pytest_warns_bounded(
+            FutureWarning,
+            "DataFrameGroupBy.apply operated on the grouping columns.",
+            lower="2.2.99",
+            upper="2.3.99",
+        ),
```

Comment:
same comment - just leave in the 2.3 test since that is all we are testing against